### PR TITLE
Extract language codes into object

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Language picker

### Description
This PR extracts the supported languages into an array

### Steps

- [x] added implementation
- [x] added / updated tests
- [x] added / updated documentation
- [x] extended changelog

### CODE REVIEW

File: src/components/landing/layout/footer/language-picker.tsx [hunk 1]
```diff
@@ -1,62 +1,68 @@
...
     await i18n.changeLanguage(language)
-  }
-
-  const shortenLanguageCode = (language: string): string => {
-    const languageWithoutArea = language.substr(0, 2)
-    if (languageWithoutArea === 'zh') {
-      return language
-    } else {
-      return languageWithoutArea
-    }
```
Comment (DerMolly) [inline src/components/landing/layout/footer/language-picker.tsx:21]: brauchen wir den AreaCode nicht mehr?

Comment (ErikMichelson) [reply src/components/landing/layout/footer/language-picker.tsx:21]: Mit dem neuen Code wird direkt der Testwert vom Browser verglichen einmal mit den Keys der Liste direkt und einmal der Substring ohne Area-Code mit den Keys der Liste.

Comment (ErikMichelson) [reply src/components/landing/layout/footer/language-picker.tsx:21]:
```javascript
return wantedLanguage === supportedLanguage || wantedLanguage.substr(0, 2) === supportedLanguage
```

Comment (mrdrogdrog) [reply src/components/landing/layout/footer/language-picker.tsx:21]: We don't need this specific case anymore. I made it a bit more general.

Comment (mrdrogdrog) [reply src/components/landing/layout/footer/language-picker.tsx:21]: guys, english please

File: src/components/landing/layout/footer/language-picker.tsx [hunk 2]
```diff
@@ -1,62 +1,69 @@
...
+  cs: 'Česky',
+  sk: 'Slovensky'
+}
+
+const findLanguageCode = (wantedLanguage: string): string => {
+  let foundLanguage = Object.keys(languages).find((supportedLanguage) => wantedLanguage === supportedLanguage)
+  if (!foundLanguage) {
+    foundLanguage = Object.keys(languages).find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
+  }
+  return foundLanguage || ''
```
Comment (SISheogorath) [inline src/components/landing/layout/footer/language-picker.tsx:42]: How about:

```
return languages[wantedLanguage] || languages[wantedLanguage.substr(0, 2)] || ''
```

That would save two O(f(n))¹ functions and replace it with an O(1) function.

¹ f(n) is the search functions complexity for n.

Comment (mrdrogdrog) [reply src/components/landing/layout/footer/language-picker.tsx:42]: Well, valid point. Feel free to open a PR :yum:

**Pull request merged**